src/yul_parser.rs

```diff
@@ -57,15 +71,27 @@ impl Identifier {
 
 impl Literal {
     fn from(pair: Pair<Rule>) -> Literal {
+        let span = pair.clone().into_span();
+        let location = Some(SourceLocation {
+            start: span.start(),
+            end: span.end(),
+        });
```

Comment:
I'd rather use [`as_span()`](https://docs.rs/pest/latest/pest/iterators/struct.Pair.html#method.as_span) here because it does not consume the pair but it's only available in pest 2. `into_span()` is actually deprecated in that version. The project still depends on pest 1 though.